apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Types/MessageArchive/ArchivedMessageSortInput.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.WebApi.GraphQL.Enums;
+
+namespace Energinet.DataHub.WebApi.GraphQL.Types.MessageArchive;
+
+public record ArchivedMessageSortInput(
+    SortDirection? MessageId,
+    SortDirection? DocumentType,
+    SortDirection? Sender,
+    SortDirection? Receiver,
+    SortDirection? CreatedAt);
```

Comment:
I think we can drop some code, if this look like this instead

```suggestion
public record ArchivedMessageSortInput(
    SortDirection? SortDirection,
    FieldToSortBy? FieldToSortBy);
```

Reply:
Yes I agree, that would be a simpler solution 👍 But there is a reason I'm not doing that. Long story short, the structure of this SortInput class follows the same convention as Hot Chocolate when you use the `UseSorting` attribute (https://chillicream.com/docs/hotchocolate/v13/fetching-data/sorting). This is because, in the frontend, we have code that can automatically add server-side paging with sorting to any table, provided that the underlying query follows the same convention. In other words, if you were to add `UsePaging` and `UseSorting` to any query that returns a list, we have a setup that automatically generates a class for the frontend to use in a table component ([this is all it takes to make a table server-side paginated with sorting](https://github.com/Energinet-DataHub/greenforce-frontend/blob/daa5f98ad9e59a9e8522d0493c4035799a6e3bac/libs/dh/message-archive/feature-search/src/lib/table.component.ts#L130-L136)). AFAIK there is no way to alter the sort input type convention of `UseSorting`, so I'd rather just go with that over having to support multiple conventions 🙂